components/Buttons/src/MDCFloatingButton.h

```diff
@@ -52,6 +60,20 @@ typedef NS_ENUM(NSInteger, MDCFloatingButtonShape) {
 @interface MDCFloatingButton : MDCButton
 
 /**
+ The shape of this floating button.
+
+ Changing this value will effect how the title and image are positioned.
+ */
+@property(nonatomic, assign) MDCFloatingButtonShape shape;
```

Comment:
The requirement to be able to change from Default to Extended opens the API to switch to and from any combination of the other shapes.